### ISSUE TITLE
nemini on Windows

### ISSUE THREAD

**Issue by lm8** [neonmoe/nemini#1]
Was able to build nemini on Windows with relatively little patching.  I'd tried building Lagrange before and had issues with building it with the compiler from mingw.org (not to be confused with the mingw64 compiler).  I'm using libressl.  Was able to view gemini pages.  However, some don't fit within the screen and I'm at a loss as to how you're supposed to scroll to see the rest of the page.  The only SDL events I could find in the code were mouse related.  I have a touchpad and no mouse on the system I'm using.  Was wondering if there was an easy way to add keyboard arrow key support to try to view the parts of the page that are missing.

Really nice job on this.  It's the only Gemini client I've been able to build from source so far that works cross-platform on systems like Windows and I've tried to build more than one Gemini client.

Thanks.

**Comment by lm8** [comment]
Okay, some quick and dirty code just to get some keys scrolling is below.  However, it doesn't have bounds checking.  Probably, adding something for pageup and pagedown would be nice too. 

             else if (event.type == SDL_KEYDOWN) {
                if (event.key.keysym.sym == SDLK_UP)
                {   
                float scale = 0;
                get_scale(window, renderer, &scale, NULL);
                float line_height = text_line_height(scale);
                scroll_delta += line_height * 10;
                }   
                else if (event.key.keysym.sym == SDLK_DOWN)
                {   
                float scale = 0;
                get_scale(window, renderer, &scale, NULL);
                float line_height = text_line_height(scale);
                scroll_delta -= line_height * 10;
                }   
            }

**Comment by lm8** [comment]
That should have if (scroll_delta < 0)

**Comment by neonmoe** [comment]
Thanks for the kind words! I had not anticipated keyboard-only use, so this is indeed a current shortcoming, thanks for reporting. Seems like a relatively easy fix, so I'll see if can find some time for it this week.

**Comment by neonmoe** [comment]
No bounds checking needed actually, since scroll_delta is clamped later in the code (if that's what you meant). Added arrows+page up/down+home/end in 40e85e5ccc6225180880a506f1452512b27e3b1e. However, not all pages seem to load, so there's probably been a change to the spec, or perhaps Nemini just has more bugs than I remembered :smile: Those will need more in-depth debugging, and there's no history implemented yet, so this client does still need a little bit of work to be a daily gemini driver.

**Comment by lm8** [comment]
Nice addition on the key-handling.  Thank you very much. 

Noticed there wasn't a way to go back to a previous page.  I usually use alt-left arrow to go back a page and alt-right arrow to return to a page if I backed up to a previous page and wanted to get back to the page I was at before I backed up.  Not sure how you'd do it with just a mouse stroke (unless you added some controls like back and forward arrows to a page).  Some sort of history would definitely be a nice feature.  Should be easy enough to store the last x (pick a number) URLs (or whatever they call the addresses in Gemini) and move between them with the alt arrow keys.  

If I can help at all in debugging or adding features, let me know.

I really like your implementation of a Gemini browser and how portable you've made it.   I'm sure it would work on a variety of platforms.
